src/examples/generation.ts

```diff
@@ -0,0 +1,100 @@
+import { Armature } from '../armature/Armature';
+import { Node, Point } from '../armature/Node';
+import { genSphere } from '../geometry/Sphere';
+import { Light } from '../renderer/interfaces/Light';
+import { Renderer } from '../renderer/Renderer';
+
+import { mat4, quat, vec3 } from 'gl-matrix';
+import { flatMap } from 'lodash';
+
+const light1: Light = {
+    lightPosition: [10, 10, 10],
+    lightColor: [0.3, 0.3, 0.3],
+    lightIntensity: 256
+};
+const light2: Light = {
+    lightPosition: [700, 500, 50],
+    lightColor: [0.3, 0.3, 0.3],
+    lightIntensity: 100
+};
+
+const renderer: Renderer = new Renderer(800, 600, 2, vec3.fromValues(0.2, 0.2, 0.2));
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Step 1: create geometry
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Add lights to the renderer
+renderer.addLight(light1);
+renderer.addLight(light2);
+
+const leafSphere = genSphere();
+leafSphere.colors = Float32Array.from(flatMap(leafSphere.vertices, () => [0.8, 1, 0.8]));
+
+const branchSphere = genSphere();
+branchSphere.colors = Float32Array.from(flatMap(branchSphere.vertices, () => [0.4, 0.3, 0.3]));
```

Comment:
I'm having a hard time understanding this, even though I know this is for setting the colours and I *think* the numbers represent RGB. Do we want to abstract this for the future?

Reply:
To clarify, is it just the "magic number" array that is confusing here? I think an abstraction for colour would be useful (could also potentially let us do more converting behind the scenes between colour systems (RGB, CMYK, etc.).

Reply:
No I found almost everything beyond the equal sign confusing. The magic numbers was the only part I found not confusing initially because I understood that's where the actual colour part is coming from. 

After reading it again I'm thinking that it's mapping every vertex to the same colour and then creating a new array to assign to `colors`. Correct me if I'm wrong?

Reply:
Yeah, giving each vertex the same colour, and packing into a flat array. We have an issue open to make a colour API, but none exists yet, so this is a temporary thing to hack in colours until we do.

Reply:
The user could do the same thing using a conventional for loop:
```
const numVertecies = branchSphere.vertices.length;
for (var i = 0; i < numVertcies; i += 1) {
    branchSphere.colors.push(0.4, 0.3, 0.3);
}
```